apphub/automl/rand_augment/rand_augment_tf.py

```diff
@@ -0,0 +1,310 @@
+        shear_coeff = self.shear_coef * random.choice([1.0, -1.0])
+        width, height = im.size
+        xshift = int(round(self.shear_coef * width))
+        new_width = width + xshift
+        im = im.transform((new_width, height),
+                          ImageTransform.AffineTransform(
+                              (1.0, shear_coeff, -xshift if shear_coeff > 0 else 0.0, 0.0, 1.0, 0.0)),
+                          resample=Image.BICUBIC)
+        if self.shear_coef > 0:
+            im = im.resize((width, height))
+        return np.copy(np.asarray(im))
+
+
+class ShearY(NumpyOp):
+    def __init__(self, level, inputs=None, outputs=None, mode=None):
+        super().__init__(inputs=inputs, outputs=outputs, mode=mode)
+        self.shear_coef = level * 0.03
+
+    def forward(self, data, state):
+        im = Image.fromarray(data)
+        shear_coeff = self.shear_coef * random.choice([1.0, -1.0])
+        width, height = im.size
+        yshift = int(round(self.shear_coef * height))
+        newheight = height + yshift
+        im = im.transform((width, newheight),
+                          ImageTransform.AffineTransform(
+                              (1.0, 0.0, 0.0, shear_coeff, 1.0, -yshift if shear_coeff > 0 else 0.0)),
+                          resample=Image.BICUBIC)
+        if self.shear_coef > 0:
+            im = im.resize((width, height))
+        return np.copy(np.asarray(im))
+
+
+class TranslateX(NumpyOp):
+    def __init__(self, level, inputs=None, outputs=None, mode=None):
+        super().__init__(inputs=inputs, outputs=outputs, mode=mode)
+        self.level = level
+
+    def forward(self, data, state):
+        im = Image.fromarray(data)
+        width, height = im.size
+        displacement = int(self.level / 10 * width / 3 * random.choice([1.0, -1.0]))
+        im = im.transform((width, height),
+                          ImageTransform.AffineTransform((1.0, 0.0, displacement, 0.0, 1.0, 0.0)),
+                          resample=Image.BICUBIC)
+        return np.copy(np.asarray(im))
+
+
+class TranslateY(NumpyOp):
+    def __init__(self, level, inputs=None, outputs=None, mode=None):
+        super().__init__(inputs=inputs, outputs=outputs, mode=mode)
+        self.level = level
+
+    def forward(self, data, state):
+        im = Image.fromarray(data)
+        width, height = im.size
+        displacement = int(self.level / 10 * height / 3 * random.choice([1.0, -1.0]))
+        im = im.transform((width, height),
+                          ImageTransform.AffineTransform((1.0, 0.0, 0.0, 0.0, 1.0, displacement)),
+                          resample=Image.BICUBIC)
+        return np.copy(np.asarray(im))
+
+
+def get_estimator(level,
+                  num_augment,
+                  epochs=24,
+                  batch_size=512,
+                  max_train_steps_per_epoch=None,
+                  max_eval_steps_per_epoch=None):
+    assert level <= 10, "the level should not be more than 10"
```

Comment:
do we want to assert 0 <= level <= 10 to make it more clear?

Reply:
done